templates/items/backpack.hbs

```diff
@@ -51,6 +51,16 @@
                 </label>
             </div>
 
+            {{!-- Container Properties --}}
+            <div class="form-group stacked container-properties">
+                <label>{{ localize "DND5E.ItemContainerProperties" }}</label>
+                {{#each properties}}
+                <label class="checkbox">
+                    <input type="checkbox" name="system.properties.{{@key}}" {{checked selected}}> {{ label }}
+                </label>
+                {{/each}}
+            </div>
```

Comment:
Then if we don't add `weightless` to the properties, how about we still show it in this list to avoid too many large sections with just a single checkbox:

```suggestion
            <div class="form-group stacked">
                <label>{{localize 'DND5E.ItemContainerProperties'}}</label>
                {{#each properties}}
                <label class="checkbox">
                    <input type="checkbox" name="system.properties.{{@key}}" {{checked selected}}> {{ label }}
                </label>
                {{/each}}
                <label class="checkbox">
                    <input type="checkbox" name="system.capacity.weightless" {{checked system.capacity.weightless}}>
                    {{localize 'DND5E.ItemContainerWeightless'}}
                </label>
            </div>
```

Reply:
I've gone ahead with an alternative that is also much more in line with how we handle these related inputs on other item sheets, merging all capacity-related fields into one row. :)

![image](https://github.com/foundryvtt/dnd5e/assets/50169243/a5425e1e-377e-4775-ab75-bb8ebe104b19)

Reply:
I'm not so sure, because "Weightless Contents" doesn't really have to do with the carrying capacity of the backpack.

Also, by just changing it to "Weightless" it isn't clear that that only applies to the contents and not the whole container.

Reply:
We may be splitting hairs but I'm not sure I agree; all three inputs are related to the contents of the container; how much, what kind, and if it weighs.

We could relabel the form group from 'Capacity' to 'Contents'.

Reply:
Then we'd lose the clarity that those values represent the maximum container capacity.

Time to call in a 3rd party! @Fyorl Any thoughts?

Reply:
Lives may be at stake.

Reply:
A second idea I had was to keep it the way it originally was - the issue seemed to be that the form group for container contents had only a single checkbox. We could, if need be, come up with more properties to fill it with. I'm sure future backpack management features could make use of some.

Reply:
Feels like it should be part of the container's `properties` to me. What's the argument against that? That it's specific to containers? Aren't most weapon properties also only specific to weapons?

Reply:
We (or at least I) do not appear able to migrate it. See here: https://github.com/foundryvtt/dnd5e/pull/2709#discussion_r1439092290

Reply:
Do you foresee any issues with the following migration?
```js
#migrateWeightless(source) {
  if ( foundry.utils.getProperty(source, "capacity.weightless") === true ) {
    source.properties ??= [];
    source.properties.push("weightless");
  }
}
```

Reply:
I may be incorrect, but would that not cause an issue if doing `item.update({"system.capacity.weightless": true})` regardless of if the property exists?

Reply:
Yes, it would wipe all the other properties (in this case there's only one other property to wipe). But our system code will not be performing such updates, and if there's a module doing that, then it's really no different to any number of problems an out-of-date module might cause.

If we wanted a more robust deprecation path, I suppose we could perform this update as part of `migration.mjs`, rather than an in-memory migration (with all of the associated caveats that entails).

Reply:
Haven't fully tested this, but it seems like we could just do this in `migrateItemData`:

```javascript
  // Migrate container properties
  if ( (item.type === "backpack") && (foundry.utils.getProperty(item, "system.capacity.weightless") === true) ) {
    updateData["system.properties"] = foundry.utils.getProperty(item, "system.properties") ?? [];
    updateData["system.properties"].push("weightlessContents");
  }
```

Reply:
Okay, that one didn't fully work, but here is a diff for a change that uses a similar system to how we handled the `TraitAdvancement` migration on classes.

[changes.diff.txt](https://github.com/foundryvtt/dnd5e/files/13814688/changes.diff.txt)

Reply:
Can you elaborate on why the solution mentioned here didn't work: https://github.com/foundryvtt/dnd5e/pull/2709#discussion_r1439942453

I think I'd prefer it over the solution presented in the diff which feels overly complicated for what should be a very simple migration that we have done many times before in `migration.mjs`.